tests/020-module.test.js

```diff
@@ -0,0 +1,33 @@
+const mock = require('mock-fs');
+const serverModule = require('../');
+
+describe('Test module export', () => {
+    it('should return an error object because no file', async () => {
+        // mock file in temporary file system
+        mock({
+            '/proc/': {
+                'mdstat': null
+            }
+        });
+        // apply module function
+        const res = await serverModule();
+        expect(res).toEqual({error: 'kernel module not loaded'});
```

Comment:
Why doesn't the module "just" throw in this case? It seems to be a faulty configuration, meaning it's for an admin to solve anyway, and adding one's own "error system" here only increases the amount of cases that have to be supported in the frontend/client...

Reply:
Hmm, but I think it isn't nice to throw an unhandled exception the "simple-server" and forget it.
So we must implement an exception handling somewhere else.

More discussion required.

Reply:
@fussel178 That will be up to the implementation of `server-base` to handle. What I'd do here is `throw new Error('kernel module not loaded')`, making it semantically correct (it is an error, after all, and the function doesn't return the expected object type), meaning you still have an error message that the `server-base` can

- log into a log file
- log to the console
- potentially send in the response in some way (TBD: would this pose potential security risks?)

Simply returning an object again means that there is no way for the server backend to know about the error and react with a `500` (it could, after all, also be the case that the Data Structure of the module return type is an object with a field `"error"`, which could be plausible for some sort of error status module showing current errors).

The thing is that we're developing a module here. When it fails, it's only job is to tell the consumer of the module (here, `server-base`) that and potentially why it has failed. It is not up to a module to decide how to handle errors that are not fixable by the module :wink: